src/mpi/datatype/typeutil.c

```diff
@@ -182,86 +251,27 @@ int MPII_create_pairtypes(void)
     return mpi_errno;
 }
 
-static int pairtypes_finalize_cb(void *dummy ATTRIBUTE((unused)))
+int MPIR_Datatype_post_init(void)
```

Comment:
Since now we know the exact scope of these init functions, how about naming them more specificly, something like `MPIR_init_builtin_datatypes` and `MPIR_commit_pairtypes`?

Reply:
I thought of being more explicit, but ended up trying to match `MPIR_Err_init` and `MPIR_Group_init`. Maybe `MPIR_Datatype_init_predefined` and `MPIR_Datatype_commit_pairtypes`?

Reply:
Sounds OK.